src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/NewAzureDeploymentCmdlet.cs

```diff
@@ -0,0 +1,72 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    /// <summary>
+    /// Creates a new deployment.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRmDeployment", SupportsShouldProcess = true,
+        DefaultParameterSetName = ParameterlessTemplateFileParameterSetName), OutputType(typeof(PSDeployment))]
+    public class NewAzureDeploymentCmdlet : ResourceWithParameterCmdletBase, IDynamicParameters
+    {
+        [Alias("DeploymentName")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The name of the deployment it's going to create. Only valid when a template is used. When a template is used, if the user doesn't specify a deployment name, use the current time, like \"20131223140835\".")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The location to store deployment data.")]
+        [LocationCompleter("Microsoft.Resources/resourceGroups")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The deployment debug log level.")]
+        [ValidateSet("RequestContent", "ResponseContent", "All", "None", IgnoreCase = true)]
```

Comment:
It's unlikely that the set would change. What's the recommendation here? I was following New-AzureRmResourceGroupDeployment. Should I change for both?

Reply:
@Tiano2017 we recommend using `PSArgumentCompleter` since it allows for potential additions in the future and users don't notice a difference with the tab-completion

Reply:
updated.

Reply:
fixed. 

---
In reply to: [205821162](https://github.com/Azure/azure-powershell/pull/6516#discussion_r205821162) [](ancestors = 205821162)